frontend/src/App.tsx

```diff
@@ -376,7 +376,11 @@ export class App extends PureComponent<Props, State> {
 
   handlePageInfoChanged = (pageInfo: PageInfo): void => {
     const { queryString } = pageInfo
-    window.history.pushState({}, "", queryString ? `?${queryString}` : "/")
+    const targetUrl =
+      document.location.origin + document.location.pathname + queryString
+        ? `?${queryString}`
+        : "/"
```

Comment:
Two things to note about these lines:
* It's fine to use relative URLs with `pushState`, so I think we'll want to do that and thus drop `document.location.origin` here for conciseness.
* This will always evaluate to `?${queryString}` since by javascript operator precedence, this is equivalent to
```javascript
(document.location.origin + document.location.pathname + queryString) ? `?${queryString}` : "/"
```

I believe we want to do something like 
```javascript
const targetUrl = document.location.pathname + (queryString ? `?${queryString}` : "")
```
where we can drop the `/` in the ternary operator as it'll come from `pathname`.